hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -728,109 +735,135 @@ protected Object putInternal(Data key, Object value, long ttl, boolean loadFromS
 
     @Override
     public Boolean merge(SplitBrainMergeEntryView<Data, Object> mergingEntry, SplitBrainMergePolicy mergePolicy) {
+        throw new UnsupportedOperationException("Use merge(SplitBrainMergeEntryView<Data, Object> mergingEntry,"
```

Comment:
Can we have the `RecordStore` not `extends SplitBrainAwareDataContainer<Data, Object, Boolean>` instead? Seems to work and will produce compile time errors (instead of runtime).

Reply:
I intend to remove `SplitBrainAwareDataContainer` completely, since we don't really need it an it enforces us to a very strict merge method interface, which is not suitable for Map, AtomicLong and AtomicReference (I cheated a bit there by misusing the key field of the merging entry). So I'm totally fine with removing it here already.

Reply:
Here is the PR to remove the interface: https://github.com/hazelcast/hazelcast/pull/12322